### PR TITLE
w changes

### PR DESCRIPTION
**Pull request by aridavidson001**
I cleaned up the pivot again
also added JoystickUtil

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/pivot/PivotIOTalonFX.java [hunk 1]
```diff
@@ -132,22 +132,12 @@ public void updateInputs(PivotIOInputs inputs) {
     inputs.followerSupplyCurrentAmps = followerPivotMotor.getSupplyCurrent().getValueAsDouble();
```
Comment (Ishan1522) [inline src/main/java/frc/robot/subsystems/pivot/PivotIOTalonFX.java:132]: remove the docstring from this method

File: src/main/java/frc/robot/subsystems/pivot/PivotIO.java [hunk 2]
```diff
@@ -32,12 +37,32 @@ public default boolean isPivotWithinAcceptableError() {
...
 
+  /**
+   * Sets the voltage of the pivot motors
+   *
+   * @param volts the voltage
+   */
   public default void setVoltage(double volts) {}
 
+  /**
+   * sets the pivot angle(rotations) shooter
```
Comment (Ishan1522) [inline src/main/java/frc/robot/subsystems/pivot/PivotIO.java:55]: capitalize docstring. I wouldn't include the unit in parenthisis as they are defined in the parameter.

File: src/main/java/frc/robot/subsystems/pivot/PivotIO.java [hunk 3]
```diff
@@ -32,12 +37,32 @@ public default boolean isPivotWithinAcceptableError() {
...
+   *
+   * @param angle the desired angle in rotations
+   */
   public default void setPivotAngle(double angle) {}
 
+  /**
+   * Gets the target angle of the pivot in degrees
```
Comment (Ishan1522) [inline src/main/java/frc/robot/subsystems/pivot/PivotIO.java:62]: Are these units correct?

File: src/main/java/frc/robot/subsystems/pivot/PivotIO.java (hunk 2 above)
Comment (Ishan1522) [reply src/main/java/frc/robot/subsystems/pivot/PivotIO.java:55]: also it doesn't necessarily set it to the position in rotations. You have an arbitrary double as a parameter. So if you pass in degrees you would have to convert to the required units for controlling the system (which for the real pivot is rotations).

File: src/main/java/frc/robot/subsystems/pivot/PivotIO.java [hunk 4]
```diff
@@ -24,6 +24,11 @@ public static class PivotIOInputs {
 
   public default void updateInputs(PivotIOInputs inputs) {}
```
Comment (Ishan1522) [inline src/main/java/frc/robot/subsystems/pivot/PivotIO.java:25]: make sure all your methods in the interface have docstrings

**Review by Ishan1522 — CHANGES_REQUESTED** [review]
make sure u capitalize ur docstrings

File: src/main/java/frc/robot/subsystems/pivot/PivotIO.java (hunk 2 above)
Comment (aridavidson001) [reply src/main/java/frc/robot/subsystems/pivot/PivotIO.java:55]: Should we set it in rotations? I think the sim code converts from rotations to radians or whatever it uses currently

File: src/main/java/frc/robot/subsystems/pivot/PivotIO.java (hunk 3 above)
Comment (aridavidson001) [reply src/main/java/frc/robot/subsystems/pivot/PivotIO.java]: im just gonna do everything in pivotIO as rotations

**Comment by aridavidson001** [comment]
just needs spotlessApply

**Comment by Ishan1522** [comment]
🤯 🤯 🤯 thats crazy

**Review by Ishan1522 — APPROVED** [review]

File: src/main/java/frc/robot/subsystems/pivot/PivotIO.java [hunk 5]
```diff
@@ -32,12 +42,32 @@ public default boolean isPivotWithinAcceptableError() {
     return false;
```
Comment (Ishan1522) [inline src/main/java/frc/robot/subsystems/pivot/PivotIO.java:42]: add docstring

**Pull request merged**